tests/test_iam/test_iam.py

```diff
@@ -2885,3 +2886,806 @@ def test_delete_role_with_instance_profiles_present():
+            "createDate": policy["configuration"]["policyVersionList"][1]["createDate"],
+        },
+    ]
+    assert policy["supplementaryConfiguration"] == {}
+
+
+@mock_iam
+@mock_config
+def test_policy_config_client():
+    from moto.iam.models import ACCOUNT_ID
+    from moto.iam.utils import random_policy_id
+
+    basic_policy = {
+        "Version": "2012-10-17",
+        "Statement": [{"Action": ["ec2:*"], "Effect": "Allow", "Resource": "*"}],
+    }
+
+    iam_client = boto3.client("iam", region_name="us-west-2")
+    config_client = boto3.client("config", region_name="us-west-2")
+
+    account_aggregation_source = {
+        "AccountIds": [ACCOUNT_ID],
+        "AllAwsRegions": True,
+    }
+
+    config_client.put_configuration_aggregator(
+        ConfigurationAggregatorName="test_aggregator",
+        AccountAggregationSources=[account_aggregation_source],
+    )
+
+    result = config_client.list_discovered_resources(resourceType="AWS::IAM::Policy")
+    assert not result["resourceIdentifiers"]
+
+    # Make 10 policies
+    policies = []
+    num_policies = 10
+    for ix in range(1, num_policies + 1):
+        this_policy = iam_client.create_policy(
+            PolicyName="policy{}".format(ix),
+            Path="/",
+            PolicyDocument=json.dumps(basic_policy),
+            Description="policy{}".format(ix),
+        )
+        policies.append(
+            {
+                "id": this_policy["Policy"]["PolicyId"],
+                "name": this_policy["Policy"]["PolicyName"],
+            }
+        )
+
+    assert len(policies) == num_policies
+
+    # Test non-aggregated query: (everything is getting a random id, so we can't test names by ordering)
+    result = config_client.list_discovered_resources(
+        resourceType="AWS::IAM::Policy", limit=1
+    )
+    first_result = result["resourceIdentifiers"][0]["resourceId"]
+    assert result["resourceIdentifiers"][0]["resourceType"] == "AWS::IAM::Policy"
+    assert len(first_result) == len(random_policy_id())
+
+    # Test non-aggregated pagination
+    assert (
+        config_client.list_discovered_resources(
+            resourceType="AWS::IAM::Policy", limit=1, nextToken=result["nextToken"]
+        )["resourceIdentifiers"][0]["resourceId"]
+    ) != first_result
+
+    # Test aggregated query: (everything is getting a random id, so we can't test names by ordering)
+    agg_result = config_client.list_aggregate_discovered_resources(
+        ResourceType="AWS::IAM::Policy",
+        ConfigurationAggregatorName="test_aggregator",
+        Limit=1,
+    )
+    first_agg_result = agg_result["ResourceIdentifiers"][0]["ResourceId"]
+    assert agg_result["ResourceIdentifiers"][0]["ResourceType"] == "AWS::IAM::Policy"
+    assert len(first_agg_result) == len(random_policy_id())
+    assert agg_result["ResourceIdentifiers"][0]["SourceAccountId"] == ACCOUNT_ID
+    assert agg_result["ResourceIdentifiers"][0]["SourceRegion"] == "global"
```

Comment:
So in aggregated queries, the `SourceRegion` is not `global`. It's the region for where the config recorder resides. This should also be true for non-aggregated queries as well.

I think the _least offensive thing to do here_ is make this a static variable, like `ACCOUNT_ID`, which will allow users to mock that out to whatever region they want it to be.

And to add more complexity :), this will also impact the filtering for aggregated queries.

Reply:
Good catch here. I went a bit of different direction (hopefully it's not too offensive). I am trying to replicate the IAM config functionality as close as possible to real life, and getting the "duplicate" resources are an important piece I'm trying to test in my application. 

I first went about it by treating IAM differently in `ConfigBackend#list_aggregate_discovered_resources`, but that just messed up all of the pagination/tokens, so I just moved that logic into the IAM specific config queries. That required me to add a new argument (the aggregator) into the signature for `list_config_service_resources`.

Reply:
I am curious about this. You should only see duplicates of resources in your aggregator if you have multiple in-account/in-region config recorders that record the same resource (and aggregating it in 1 spot).

For example:
If you have a config recorder in us-east-1, us-west-2, and eu-west-1 that are all configured to record IAM resources, AND those recorders are being aggregated by the aggregator `foo`, then when you query against `foo` you will see **3** results: one for each config region.

Reply:
^^ And this only applies to Global resources. For regional resources, the aggregator will list all resources regardless of the region per the note we left here https://github.com/spulec/moto/blob/master/moto/core/models.py#L706-L710.

IAM is an edge case here as it's unlike most AWS resource types :)

Reply:
It might be easier for you to change the pagination logic to make the token a concatenation of the `role_id + config_recorder_region`.

Reply:
@mikegrima Sorry - I didn't even see these responses. I should look closer next time!

I agree with your example and had a slight misunderstanding on how things worked. Of course, this now poses even more complexity. IAM returned aggregated resources are now to depend on:

- the existing list criteria already built in moto (filters, aggregation) 
- the recorder status (ie is the recorder running)
- the recorder setup (ie is it recording the resource type you want, as well is it a part of includeGlobalResourceTypes)

Also, I am not sure what "enables" the IAM policies to show up in the recorder, because if you have an "mostly untouched/empty" region, the IAM resources don't show up in both non-aggregated or aggregated queries. Phrased differently...I have 4 region-specific recorders running in my test AWS account, where two of the regions have "stuff" (instances, lambda functions, etc), but two don't. I only see the roles and policies in the regions where I have "stuff". Granted, I've only had those recorders "empty region" recorders on for ~4 hours, so maybe it's an issue with propagation. I do see 8 resources (default vpc's, etc) in those "empty regions". I'll check back in 24 hours to see if they show up.

I am not sure I follow your suggestion to mock out the recorder variable. It seems like I should instead just implement the resource config queries to read from the config recorders off the moto backend. I'd like to solidify the approach before I dive in. Does building "native" support for recorders sound like a good idea?
 
Also, apologies for all the back and forth; this is a learning process for me!

Reply:
> the existing list criteria already built in moto (filters, aggregation)
the recorder status (ie is the recorder running)
the recorder setup (ie is it recording the resource type you want, as well is it a part of includeGlobalResourceTypes)

When I initially developed the Config feature, I didn't make it a requirement to pre-establish a recorder. I only made the requirement to pre-establish the aggregator for aggregated queries, as AWS Config will raise an exception if one is not existing. 

As such, now that I'm thinking of it... Maybe it _does_ make sense to just have it return all the regions as duplicates, in aggregated queries (since we assume that for the aggregator, it will query all the regional backends, which we document here: https://github.com/spulec/moto/blob/master/CONFIG_README.md#how-does-this-work). For non-aggregated queries, there won't be any duplicates, so that won't be an issue - we just assume that there is a recorder in that region recording that resource type.

Reply:
If this makes sense then your PR should basically be done.

Reply:
sorry for the delay; that sounds good to me!

Reply:
I think the only thing left to do is get the unit tests to pass.